mc2mc/internal/client/odps.go

```diff
@@ -109,6 +110,28 @@ func (c *odpsClient) GetOrderedColumns(tableID string) ([]string, error) {
 	return columnNames, nil
 }
 
+// generateLogView generates the log view for the given task instance
+func (c *odpsClient) generateLogView(taskIns *odps.Instance) (string, error) {
+	u, err := c.client.LogView().GenerateLogView(taskIns, c.logViewRetentionInDays*24)
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+
+	// change query parameter h to http://service.id-all.maxcompute.aliyun-inc.com
+	parsedURL, err := url.Parse(u)
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+	q := parsedURL.Query()
+	q.Set("h", "http://service.id-all.maxcompute.aliyun-inc.com/api")
```

Comment:
Have we verified the result, whether it is safe to replace host, which includes scheme and API path?

Reply:
yes, it has been verified. it's safe as we're expecting that the logview could be accessible from public endpoint